apollo-federation/src/operation/mod.rs

```diff
@@ -2596,7 +2666,9 @@ impl SelectionSet {
     /// Removes the @defer directive from all selections without removing that selection.
     fn without_defer(&mut self) {
         for (_key, mut selection) in Arc::make_mut(&mut self.selections).iter_mut() {
-            Arc::make_mut(selection.get_directives_mut()).retain(|dir| dir.name != name!("defer"));
+            // TODO(@goto-bus-stop): doing this changes the key of the selection!
+            // We have to rebuild the selection map.
```

Comment:
I didn't fix this here but this is actually a bug in `without_defer`: it creates invalid outputs. `get_directives_mut()` is not a method that should exist on selection types because directives affect the key, and you're not meant to be able to change things about the selection that affect the key.

Reply:
good to know for when we do defer!

Reply:
Well, this is actually more of a problem while we don't have defer 😛